lib/plugins/threaded-signing/plugin.c

```diff
@@ -254,30 +279,34 @@ threaded_openssl_get_signature(sv_threaded_plugin_t *self,
   SignedVideoReturnCode status = SV_OK;
 
   g_mutex_lock(&self->mutex);
-  if (self->plugin_state == THREADED_SIGNING_HAS_SIGNATURE && self->signature_info) {
-    if (self->signature_info->signature_size > max_signature_size) {
+  if (self->output_buffer[0].signature) {
+    if (self->output_buffer[0].size > max_signature_size) {
       // If there is no room to copy the signature, report zero size.
       *written_signature_size = 0;
     } else {
-      memcpy(signature, self->signature_info->signature, self->signature_info->signature_size);
-      *written_signature_size = self->signature_info->signature_size;
+      // Get the oldest signature
+      memcpy(signature, self->output_buffer[0].signature, self->output_buffer[0].size);
+      *written_signature_size = self->output_buffer[0].size;
+      // Free |signature| and move buffer
+      const int output_buffer_end = self->output_buffer_idx;
+      free(self->output_buffer[0].signature);
```

Comment:
Instead of freeing the memory you should reuse the same slot if you allocate `max_signature_size`.